src/vision/status_quo/frederik_makes_their_first_steps_into_async.md

```diff
@@ -0,0 +1,63 @@
+
+### Learning about Async
+Still enarmored with the first edition of the Rust book, they decide to go looking for an updated version, hoping that it will teach them async in the same manner that it taught them so much about the language and design patterns for Rust. Dissapointed, they find no mention of async in the book, aside from a note that it exists as a keyword.
+
+Not to be deterred, they go looking further, and start looking for similarly great documentation about async.
+After stumbling upon the async book, their dissapointment is briefly replaced with relief as the async book does a good job at solidifying what they have already learned in various blog posts about async, why one would use it and even a bit about how it all works under the hood.
+They skim over the parts that seem a bit too in depth for now like pinning, as they're looking to quickly get their hands dirty.
+Chapter 8: The Async Ecosystem teaches them what they already picked up on through blog posts and contentious tweets: the choice of the runtime has large implications on what libraries they can use.
+
+### The wrong time for big decisions
+Frederik's dreams to quickly get their hands dirty with async Rust are shattered as they discover that they first need to make a big choice: what executor to use. Having had quite a bit of exposure to the conversations surrounding the incompatible ecosystems, Frederik is perhaps a bit more paranoid about making the wrong choice than the average newcomer.
+This feels like a big decision to them, as it would influence the libraries they could use and switching to a different ecosystem would be all but impossible after a while. Since they would like to choose what libraries they use before having to choose an executor, Frederik feels like the decision making is turned on its head. 
+
+Their paranoia about choosing the right ecosystem is eased after a few days of research, and some more conversations on the Rust subreddit, after which they discover that most of the RPC libraries they might want to use are situated within the most popular Tokio ecosystem anyways. Tokio also has a brief tutorial, which teaches them some basic concepts within Tokio and talks a bit more about async in general.
+
+### Woes of a newcomer to async
+Being reasonably confident in their choice of ecosystem, Frederik starts building their distributed system.
+After a while, they want to introduce another networking library of which the api isn't async. Luckily Frederik picked up on that blocking was not allowed in async (or at least not in Tokio), through reading some blog posts about async. More reddit discussions point them towards spawn\_blocking in Tokio, and even rayon. But they're none the wiser about how to apply these paradigms in a neat manner.
```

Comment:
I only skimmed this, but the point about "at least not in Tokio" makes me a bit sad, because there's so much stuff written about async-std trying to implement a solution that allows blocking, but they never ended up doing it, and now everyone still thinks they allow it.

Reply:
Yea, I mention that not because I'm under the impression that they allow it, but because I know it might some day be possible. It might well never happen, but since I'm not educated enough to ascertain that blocking in async will forever be impossible, I didn't want to state it as a certainty.

I mention that specific blog post later in my sources.

Reply:
I changed the wording to `or at least not in any of the currently existing executors` to avoid confusion :)

Reply:
In this spirit of stories being personal experience and thus they cannot be _wrong_ only _misinformed_, it might be interesting to throw this into a story about mixing blocking into the async world. I, however, have already suggested we separate the blocking vs non-blocking aspects of this story into a separate story so I wouldn't add that to this story per say.

Reply:
I didn't really intend for this story to be about blocking, but more about the uncertainty that starting in async brings with it. Both surrounding ecosystem choices, but also surrounding design patterns. Design patterns for inclusion of sync code is just one particular example here. This story is really about the need for an authoritative path for newcomers, which provides them with a solid foundation to explore the ecosystem with, and less about any specific technical problems surrounding blocking.